### PR TITLE
Fix use case when changing status for already existed alias

### LINKED ISSUE #37676: Create an second alias with the same name and another status doesnt' change the statut
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

If you add 2 aliases with the same name, it is the first one that is entered that is taken into account in choosing its status.

https://www.loom.com/share/50a15474cc3641eab0c815fee9108d9f?sid=29fa57ff-f35a-4676-ac84-7b42037d5876

### Expected behavior

Create a second alias with another status will change the status.

### Steps to reproduce

1. Shop Parameter > Search
2. Add new alias
3. Search name = Test1
4. Alias = Test
5. Test enabled
6. Save
7. Edit Test1
8. add Alias = Test
9. Test disabled
10. See error

### PrestaShop version(s) where the bug happened

9.0.x

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by boherm**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x
| Description?      | Fix use case when changing status for already existed alias
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See #37676 
| UI Tests          | https://github.com/boherm/ga.tests.ui.pr/actions/runs/12358462822
| Fixed issue or discussion?     | Fixes #37676
| Related PRs       | ~
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Review by matks — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]
Thanks @boherm

**Comment by paulnoelcholot** [comment]
Hello @boherm,

I tested your PR and it's good for me! :tada: 

Thanks!

**Review by paulnoelcholot — DISMISSED** [review]

File: tests/Integration/Behaviour/Features/Scenario/Search/add_alias.feature [hunk 1]
```diff
@@ -46,6 +46,21 @@ Feature: Add basic alias from Back Office (BO)
       | small   | false  |
       | so biig | true   |
 
+  Scenario: I add alises for one search term for the second time with different active status
```
Comment (FabienPapet) [inline tests/Integration/Behaviour/Features/Scenario/Search/add_alias.feature]:
```suggestion
  Scenario: I add aliases for one search term for the second time with different active status
```

File: src/Adapter/Alias/Repository/AliasRepository.php [hunk 2]
```diff
@@ -90,7 +90,11 @@ public function addAliases(string $searchTerm, array $aliases): array
 
         foreach ($aliases as $searchAlias) {
             // As search term is not a primary key, we need make sure that alias and search combination does not exist
-            if ($this->aliasExists($searchAlias['alias'], $searchTerm)) {
+            // if alias exists for search term, we need to apply new active if needed
+            $aliasIfExist = $this->getAliasIfExist($searchAlias['alias'], $searchTerm);
```
Comment (FabienPapet) [inline src/Adapter/Alias/Repository/AliasRepository.php]:
```suggestion
            $aliasIfExists = $this->getAliasIfExist($searchAlias['alias'], $searchTerm);
```

File: src/Adapter/Alias/Repository/AliasRepository.php [hunk 3]
```diff
@@ -129,9 +133,9 @@ public function get(AliasId $aliasId): Alias
      * @param string $alias
      * @param string $searchTerm
      *
-     * @return bool
+     * @return Alias|null
      */
-    public function aliasExists(string $alias, string $searchTerm): bool
+    public function getAliasIfExist(string $alias, string $searchTerm): ?Alias
```
Comment (FabienPapet) [inline src/Adapter/Alias/Repository/AliasRepository.php]: Renaming a public method is BC break, + there's a typo

```suggestion
    public function getAliasIfExists(string $alias, string $searchTerm): ?Alias
```

**Review by FabienPapet — DISMISSED** [review]

File: src/Adapter/Alias/Repository/AliasRepository.php (hunk 3 above)
Comment (boherm) [reply src/Adapter/Alias/Repository/AliasRepository.php]: For the BC break, yes and no! In fact, this function is written in 2023 by Darius when the migration of Aliases has been started. But, this migration wasn't really done and although this was behind a feature flag, this first atempt of migration was not usable at all. For this 9.0.x version, isn't a big deal to change this signature.

Comment (FabienPapet) [reply src/Adapter/Alias/Repository/AliasRepository.php]: ok

**Comment by paulnoelcholot** [comment]
Than you @FabienPapet and @boherm,

I tested with the refactor and it's still good for me, I don't see any regression. The PR can be merged once the CI is green and the review is done!

**Review by nicosomb — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Pull request merged**